processing/src/main/java/org/apache/druid/query/groupby/strategy/GroupByStrategyV2.java

```diff
@@ -285,6 +342,34 @@ public boolean doMergeResults(final GroupByQuery query)
     }
   }
 
+  private void fixResultRowWithTimestampResultField(
```

Comment:
This remapping should be done in the sql layer. See `QueryMaker.remapFields()`.

Reply:
This is needed due to the time_floor dimension can be referenced by the postAgg and the nested groupBy. This is not the `QueryMaker.remapFields()` case. `QueryMaker.remapFields()` is to map the outer-most group by result to the sql level result.

Reply:
Hmm, I see that `__time` cannot be referenced by the post aggregators today, but what is the issue with the nested groupBy?

Reply:
the outer process may reference the inner groupby query's time_floor dimension

Reply:
Hmm, maybe `moveOrReplicateTimestampInRow()` is a better name.